app/assets/javascripts/components/dashboard/navigation/panels/signals/active_signal_panel.jsx

```diff
@@ -1,51 +1,46 @@
-var ActiveSignalPanel = React.createClass({
-  render: function() {
-    return (
-      <div>
-        {this.renderPanel()}
-      </div>
-    );
-  },
+import React, { Component } from 'react';
+import _ from 'lodash';
+import SignalIcon from '../../../../links/signal_icon.jsx';
 
-  handleClick: function(idx, signalType) {
+export default class ActiveSignalPanel extends Component {
+  handleClick(idx, signalType) {
     tab = {
       name: 'Edit',
       className: 'active',
-      paneId: 'new'
+      paneId: 'new',
     };
-    
+
     this.props.handleTab(tab);
     this.props.handleSignal('editSignal', this.props.signals[idx]);
-  },
-
-  renderPanel: function() {
-    var scope = this;
-    signals = [];
+  }
 
-    this.props.signals.forEach(function(s, idx) {
-      signals.push(
+  renderPanel() {
+    return this.props.signals.map((s, idx) => {
```

Comment:
This is an arrow function. It's main purpose is to bind to its surrounding scope. Thus, calling `this` within the `map` will reference the `this` of its outer scope.